native/zowed/builder.hpp

```diff
@@ -116,6 +118,15 @@ class CommandBuilder
   // Flatten a JSON object argument into the argument map (recursion not supported)
   CommandBuilder &flatten_obj(const std::string &arg_name);
 
+  // Validate both request and response using separate schemas
+  template <typename RequestT, typename ResponseT>
+  CommandBuilder &validate(bool allow_unknown_fields = false)
+  {
+    request_validator_ = std::shared_ptr<validator::ParamsValidator>(new validator::SchemaValidator<RequestT>(allow_unknown_fields));
+    response_validator_ = std::shared_ptr<validator::ParamsValidator>(new validator::SchemaValidator<ResponseT>(allow_unknown_fields));
```

Comment:
We should use `std::make_shared` here if possible - avoids use of `new` operator 😋